src/writer.rs

```diff
@@ -154,12 +154,17 @@ pub struct FdtWriterNode {
 }
 
 impl FdtWriter {
+    /// Create a new Flattened Devicetree writer instance.
```

Comment:
nit: can we update the documentation such that the descriptions of new and new_with_... are not identical?